src/script/integration/IntegrationRepository.ts

```diff
@@ -56,41 +76,45 @@ export class IntegrationRepository {
 
   /**
    * Get provider name for entity.
-   * @param {ServiceEntity|User} entity - Service or user to add provider name to
-   * @returns {Promise} - Resolves with the entity
+   * @param entity Service or user to add provider name to
+   * @returns Resolves with the entity
    */
-  addProviderNameToParticipant(entity) {
-    const shouldUpdateProviderName = entity.providerName && !entity.providerName().trim();
+  addProviderNameToParticipant(entity: ServiceEntity): Promise<ServiceEntity>;
+  addProviderNameToParticipant(entity: User): Promise<User>;
+  addProviderNameToParticipant(entity: ServiceEntity | User): Promise<ServiceEntity | User> {
+    const shouldUpdateProviderName =
+      (entity as ServiceEntity).providerName && !(entity as ServiceEntity).providerName().trim();
+
     return shouldUpdateProviderName
-      ? this.getProviderById(entity.providerId).then(providerEntity => {
-          entity.providerName(providerEntity.name);
+      ? this.getProviderById((entity as ServiceEntity).providerId).then(providerEntity => {
+          (entity as ServiceEntity).providerName(providerEntity.name);
           return entity;
         })
-      : entity;
+      : Promise.resolve(entity);
   }
 
   /**
    * Get ServiceEntity for entity.
-   * @param {ServiceEntity|User} entity - Service or user to resolve to ServiceEntity
-   * @returns {Promise} - Resolves with the ServiceEntity
+   * @param entity Service or user to resolve to ServiceEntity
+   * @returns Resolves with the ServiceEntity
    */
-  getServiceFromUser(entity) {
+  getServiceFromUser(entity: ServiceEntity | User): Promise<ServiceEntity> {
     if (entity instanceof ServiceEntity) {
       return Promise.resolve(entity);
     }
-    const {providerId, serviceId} = entity;
+    const {providerId, serviceId} = entity as any;
```

Comment:
What else is inside of `entity`? Can we type it to be `User` or `ServiceEntity`?

Reply:
`User` is not typed yet so it doesn't make sense to create an interface here.